signac/contrib/project.py

```diff
@@ -1801,7 +1803,8 @@ def keyfunction(job):
         else:
             keyfunction = key
 
-        return groupby(sorted(iter(self), key=keyfunction), key=keyfunction)
+        return groupby(sorted(iter(JobsCursor(self._project, _filter, self._doc_filter)),
```

Comment:
Why do we need to instantiate "self" again here?

Reply:
I need to use a different filter than `self.filter` (to remove Jobs which don't have that key). Instead of modifying the `self.filter`, I choose to use a different instance to

Reply:
Thx.